### DISCUSSION TITLE
[RFC] Feature request   - getting started project or video

### DISCUSSION THREAD

**Discussion by a1git** [phpMv/ubiquity#185] [Show and tell]
I want to get started in this framework, but there are no proper demos or videos on how to actually get started. 
The demo can be as simple as     creating a new controller,  making it accessible via a route, connecting to the DB and doing select,update, delete etc. 

Can someone point me to the proper documentation or video what gives an idea of how to actually get started on this.

**Comment by a1git** [comment]
php vendor/bin/Ubiquity  controller home -v
creates a file with Home 

route is /Home ( H caps ) and not /home 
so how do I get /home 


these type of practical application is missing from everywhere.

**Comment by jcheron** [comment]
I agree, there are not enough tutorials and videos.
Some stuff exists:
#### the Guide:
- [Quick start with devtools](https://micro-framework.readthedocs.io/en/latest/quickstart/quickstart.html)
- [Quick start with webtools](https://micro-framework.readthedocs.io/en/latest/quickstart/quickstart-html.html)

#### Some videos 
on Youtube: [Ubiquity youtube chanel](https://www.youtube.com/channel/UCSn_VzC0QpXCHmGDrw-_REw)

But it's imperfect (especially the videos that aren't very sexy), and in too limited number.
So there is still work to be done, and possible ways of working, for those who would like to improve it by their contributions.

#### Some demo applications

online:

- https://github.com/phpMv/todos
- https://github.com/phpMv/groups

Another repo with some samples:
- https://github.com/phpMv/ubiquity-demos


### About routes

For your capitalization problem on the routes,
if you want to access `/home`, you need to add a route :

Via annotations or attributes

```php
<?php
namespace controllers;

 use Ubiquity\attributes\items\router\Route;

 class Home extends ControllerBase{
	#[Route('/home')]
	public function index(){
		echo "route home";
	}
}
```
The cache must then be re-initialised to be effective:
```bash
Ubiquity init-cache
```

This part is documented in the [guide](https://micro-framework.readthedocs.io/en/latest/controller/router.html#static-routes)

**Comment by a1git** [comment]
This looks like a great framework, and you are putting all the hard work .. but if people who want to start cannot start, then all the hard work you and your team is doing is for nothing .. you keep on writhing this framework, but no one is able to use it. 

I suggest to maybe take one Friday off from coding and actually record your screen when you ( in english ) do a step by step demo of a basic CURD application with a  mysql backend. That should help a lot of people. 

I started with plain PHP, then moved to code igniter 3,then to symfony and now doing symfony and codeignither 4 , and trying to start on this and I struggle with the framework on what I want to do and how it works.  Your one video of a basic application with tips and tricks ( because you know this framework better then others) will really help people like me out there who want to start on this.

**Comment by jcheron** [comment]
> This looks like a great framework, and you are putting all the hard work .. but if people who want to start cannot start, then all the hard work you and your team is doing is for nothing .. you keep on writhing this framework, but no one is able to use it.
> 
> I suggest to maybe take one Friday off from coding and actually record your screen when you ( in english ) do a step by step demo of a basic CURD application with a mysql backend. That should help a lot of people.
> 
> I started with plain PHP, then moved to code igniter 3,then to symfony and now doing symfony and codeignither 4 , and trying to start on this and I struggle with the framework on what I want to do and how it works. Your one video of a basic application with tips and tricks ( because you know this framework better then others) will really help people like me out there who want to start on this.

Thanks @a1git 
I am fully aware of this

**Comment by LX-3** [comment]
https://www.youtube.com/watch?v=BKFwO6xZ3Ok

Ubiquity Tutorial videos on this french channel (captions can be auto-translated  to english), hope it can help.

Regards,
LX

**Comment by jcheron** [comment]
I know that video is a preferred support for the younger generation.
Personally, I'm from the old school. Development is not about mechanics or cooking. It is not a question of reproducing gestures, but of understanding a logic and a functioning.
And in this case, reading a good guide is worth much more than a video.

Videos have more of a marketing impact than an educational one. This does not mean that they should be totally abandoned.

**Comment by a1git** [comment]
I agree and disagree. I agree that documentation is the best and yes, we need to have good documentation. but people will look into documentation only when they are interested and see the value the framework and can get started on it to get the basics done. 
I was able to jump from pure php to codeigniter and from codeigniter to symfony beacause someone took time to create one video that shows how to get the basics right.  I have been seeing this framework on http://www.phpbenchmarks.com/en/  for a long time now ..(over a year) and I want to improve the speed of my symfony app.   One simple video is not too much to ask, when this can benefit and bring 1000s of new users to test and work and appreciate the work you have done to make this good.

**Comment by jcheron** [comment]
Well, it took me more than one Friday...
but that's probably because I'm a beginner...
So a first draft
https://www.youtube.com/watch?v=CTAXuyPf34g
Don't hesitate to give your opinion to make it better

**Comment by a1git** [comment]
It has been very useful and this is a good start.  Waiting for more :)
